qiskit_nature/problems/second_quantization/lattice/lattice_model_problem.py

```diff
@@ -0,0 +1,116 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""The Lattice Model Problem class."""
+
+from typing import Union, Callable, List, Tuple
+
+import numpy as np
+
+from qiskit.algorithms import EigensolverResult, MinimumEigensolverResult
+from qiskit_nature import ListOrDictType, settings
+from qiskit_nature.converters.second_quantization import QubitConverter
+from qiskit_nature.operators.second_quantization import SecondQuantizedOp
+from qiskit_nature.results import EigenstateResult, LatticeModelResult
+from .models.lattice_model import LatticeModel
+from ..base_problem import BaseProblem
+
+
+class LatticeModelProblem(BaseProblem):
+    """Lattice Model Problem class to create second quantized operators from a lattice model."""
+
+    def __init__(self, lattice_model=LatticeModel) -> None:
+        """
+        Args:
+            lattice_model: A lattice model class to create second quantized operators.
+        """
+        super().__init__()
+        self._lattice_model = lattice_model
+        self._main_property_name = "LatticeEnergy"
+
+    def second_q_ops(self) -> ListOrDictType[SecondQuantizedOp]:
+        """Returns the second quantized operators created based on the lattice models.
+
+        Returns:
+            A ``list`` or ``dict`` of
+            :class:`~qiskit_nature.operators.second_quantization.SecondQuantizedOp`
+        """
+        second_q_ops: ListOrDictType[SecondQuantizedOp] = self._lattice_model.second_q_ops()
+        if settings.dict_aux_operators:
+            second_q_ops = {"LatticeEnergy": second_q_ops}
```

Comment:
Ideally, the internally used model should already return `second_q_ops` in accordance with
`settings.dict_aux_operators`. But this can also be fixed later as it requires deprecation.
Nonetheless, I suggest that you use `self._main_property_name` here in order to avoid the string
duplication.